app/models/user.rb

```diff
@@ -315,17 +315,19 @@ def send_get_swapping_reminder_email
     sent_emails.create!(template: SentEmail::REMINDER_GET_SWAPPING)
   end
 
-  def send_pending_swap_reminder_email
-    # FIXME: check SentEmail to avoid duplicates
-    # return if ...
-    self.pending_swap_reminder_email = true
+  def send_pending_swap_reminder_email(other_user)
+    template = SentEmail::REMINDER_PENDING_OFFER
+    already_sent = sent_emails.where(template: template).first
+    return if already_sent
+
     save
-    UserMailer.reminder_to_accept_swap(self).deliver_now
-    sent_emails.create!(template: SentEmail::REMINDER_PENDING_OFFER)
+    UserMailer.reminder_to_accept_swap(self, other_user).deliver_now
+    sent_emails.create!(template: template)
   end
 
   def send_vote_reminder_email
     # FIXME: check SentEmail to avoid duplicates
+    # Errrm, looks like we have two ways to check if this mail was sent ?
```

Comment:
Yes, there's the old way which we should replace with this "new" `SentEmail` model, otherwise we have an ugly mess of email stuff in the `User` model.  My second commit was a totally unfinished attempt to clean that up, although we can defer that to a separate PR because not changing it for this GE.

Reply:
Filed as #982.